src/js/containers/accountLanding/AccountLandingContainer.jsx

```diff
@@ -0,0 +1,208 @@
+            }, () => {
+                this.fetchAccounts();
+            });
+        }
+    }
+
+    updateSort(field, direction) {
+        // Change sort in the state and make a new request
+        this.setState({
+            order: {
+                field,
+                direction
+            }
+        }, () => {
+            this.fetchAccounts();
+        });
+    }
+
+    showColumns() {
+        const columns = [];
+        const sortOrder = AccountsTableFields.defaultSortDirection;
+
+        AccountsTableFields.order.forEach((col) => {
+            let displayName = AccountsTableFields[col];
+            if (col === 'budgetary_resources') {
+                // Add default fiscal year to Budgetary Resources column header
+                const fy = FiscalYearHelper.defaultFiscalYear();
+                displayName = `${fy} ${displayName}`;
+            }
+            const column = {
+                columnName: col,
+                displayName,
+                defaultDirection: sortOrder[col]
+            };
+            columns.push(column);
+        });
+
+        this.setState({
+            columns
+        }, () => {
+            this.fetchAccounts();
+        });
+    }
+
+    fetchAccounts() {
+        if (this.accountsRequest) {
+            // a request is in-flight, cancel it
+            this.accountsRequest.cancel();
+        }
+
+        this.setState({
+            inFlight: true,
+            error: false
+        });
+
+        // generate the params
+        const pageSize = 50;
+        const fy = `${FiscalYearHelper.defaultFiscalYear()}`;
+        const params = {
+            sort: this.state.order,
+            page: this.state.pageNumber,
+            limit: pageSize,
+            filters: {
+                fy
+            }
+        };
+
+        this.accountsRequest = AccountLandingHelper.fetchAllAccounts(params);
+
+        this.accountsRequest.promise
+            .then((res) => {
+                this.setState({
+                    inFlight: false
+                });
+
+                this.parseAccounts(res.data);
+            })
+            .catch((err) => {
+                this.accountsRequest = null;
+                if (!isCancel(err)) {
+                    this.setState({
+                        inFlight: false,
+                        error: true
+                    });
+                    console.log(err);
+                }
+            });
+    }
+
+    parseAccounts(data) {
+        const accounts = [];
+
+        data.results.forEach((item) => {
+            // Format budgetary resources
+            const formattedCurrency =
+                MoneyFormatter.formatMoneyWithPrecision(item.budgetary_resources, 0);
+
+            let formattedAgency = item.managing_agency;
+            if (item.managing_agency_acronym) {
+                formattedAgency = `${item.managing_agency} (${item.managing_agency_acronym})`;
+            }
+
+            const account = {
```

Comment:
Can you adapt this to follow our new `BaseXXX` object approach? You can take a look in `models/v2/BaseFederalAccountAwardRow.js` and `containers/account/awards/AccountAwardsContainer.jsx` for an example.

Basically, you would create a `BaseFederalAccountLandingRow` object with a `parse` function for ingesting the API response and handle null/empty values in that function. For derived fields (such as the combined agency name + agency abbrev col), you would use getter functions to concatenate (and determine if you can, based on if abbreviation is available) on the fly.

https://gist.github.com/kevinli-work/cfb001398a15db1afa8168fcbc4b0fb1